hkmc2/shared/src/test/mlscript/basics/NewlineSels.mls

```
@@ -38,15 +38,12 @@ rcd
 :todo
 rcd
 .a
-//│ ╔══[PARSE ERROR] Expected an expression; found static selector instead
-//│ ║  l.40: 	.a
-//│ ╙──      	^^
-//│ ╔══[PARSE ERROR] Expected end of input; found static selector instead
-//│ ║  l.40: 	.a
-//│ ╙──      	^^
 //│ ╔══[WARNING] Pure expression in statement position
 //│ ║  l.39: 	rcd
 //│ ╙──      	^^^
+//│ ╔══[COMPILATION ERROR] Unexpected term form in expression position (leading dot selection)
+//│ ║  l.40: 	.a
+//│ ╙──      	^^
```

I saw this error message is raised during lowering, but I guess it's not so appropriate to do that there?

If the resolver or flow analyzer fails to resolve a leading-dot selection, shouldn't it fail immediately and report the reason why it fails, rather than silently keep the term as-is and output a vague (to me) error during lowering?

---

Good point. It should behave the same in this respect as normal selections. But that's something we're about to change anyways, so no need to do it right now.

---

though currently the resolver doesn't resolve leading-dot selections, and the flow analyzer isn't even run in this example, so the term is passed straight to lowering. We could substitute this error with a better "unresolved LDS" error by doing what you suggested below and running flow analysis in the example, although I'm not sure that the `flow` flag is appropriate in this test file, idk what guidelines you follow in terms of which tests use which flags

---

Wow so now flow analysis is an opt-in feature? Then I guess probably we want to raise an error saying that the leading dot selection is not resolved, and to resolve it we need the flow flag.

---

Ah that's rigth, flow analysis is not run here. It's all good then, carry on, nothing to see here.